File: c++/advanced/2a_fixed_size_duplicate.cpp
//Check if there exists a subarray of size <= k+1 such that it has a repeating/duplicate no. in it.

bool containsNearbyDuplicate(vector<int>& nums, int k) {
        int n = nums.size();
        int l =0;
        unordered_set<int> st;
        for(int r=0; r<n; r++){
            if(r-l > k){ 
                st.erase(nums[l]);
                l++;
            }
            if(st.find(nums[r]) != st.end()) return true;
            st.insert(nums[r]);
        }
        return false;
}

File: js/advanced/2a_fixed_size_duplicate.js
//Check if there exists a subarray of size <= k+1 such that it has a repeating/duplicate no. in it.

var containsNearbyDuplicate = function(nums, k) {
    let l = 0
    const st = new Set()
    for(let r=0; r<nums.length; r++){
        if(r-l > k){
            st.delete(nums[l])
            l++
        }
        if(st.has(nums[r])) return true
        st.add(nums[r])
    }
    return false
};